aeron/aeron.go

```diff
@@ -110,48 +114,91 @@ func (aeron *Aeron) Close() error {
 	return err
 }
 
-// AddSubscription will add a new subscription to the driver.
+// AddSubscription will add a new subscription to the driver and wait until it is ready.
+func (aeron *Aeron) AddSubscription(channel string, streamID int32) (*Subscription, error) {
+	registrationID, err := aeron.conductor.AddSubscription(channel, streamID)
+	if err != nil {
+		return nil, err
+	}
+	for {
+		subscription, err := aeron.conductor.FindSubscription(registrationID)
+		// Retry temporary error.  Any other success or error gets returned.
+		if !errors.Is(err, TemporaryError) {
+			return subscription, err
+		}
+		aeron.context.idleStrategy.Idle(0)
+	}
```

Comment:
I'm a little surprised it's okay to block for up to driverTimeoutNs (which looks like it defaults to 5s), since the other subscriptions wouldn't be getting polled during this time, but I guess if we're not getting a subscription back, that implies other problems? Either way, looks like the Java implementation does the same thing, so this looks good to me.

Reply:
This is the sync version. The client conductor is also still running in the background.

Reply:
But other subscriptions made by the application could still be starved during this time, right? Or I might be misunderstanding something.

Reply:
Yeah, you're right, I just want to make sure I understood.